components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -618,13 +618,17 @@ abstract class EngineSession(
 
     /**
      * Navigates back in the history of this session.
```

Comment:
Can we also add a note about how `userInteraction` will affect navigating back? I know it was mentioned in https://bugzilla.mozilla.org/show_bug.cgi?id=1644595#c0, but I want to make sure we consider documenting more strongly to preserve institutional knowledge about how this extra parameter will change the navigation.

Reply:
I found this https://searchfox.org/mozilla-central/source/docshell/base/nsIWebNavigation.idl#97 upon my own digging. While I agree that there is no good source of truth except for the IDL in this case, I also don't expect people to be able to navigate to the webIDL without some prior knowledge. The only "note" I was expecting was that we skip historical items with no user interaction on go back/forward to provide some context about what the parameter does. 

The key point I will only make here is to provide some context about what the parameter does, otherwise, I can imagine someone looking at it and just wondering is the difference between true vs false.

In the most extreme example for someone looking at this without context, they might git blame their way to this PR and find the Bugzilla issue, but let's assume the blame is no longer good because of some refactoring. They might do a search for [goForward](https://searchfox.org/mozilla-central/search?q=symbol:%23goForward&redirect=false), find the GV caller which has a similar doc, and continue to looking at what the JS code does which also have no comments. All in all, I just want to avoid such an exhaustive search if we can just provide a 1 liner that saids it skips items without user interaction.

Reply:
You don't need to follow the git blame, you follow the API calls that reach the idl.

> The only "note" I was expecting was that we skip historical items with no user interaction on go back/forward to provide some context about what the parameter does.

How about, "informs the engine when a navigation is requested by user gesture" or something of the likes?

Reply:
> How about, "informs the engine when a navigation is requested by user gesture" or something of the likes?

I don't know if that makes sense. I think you have to say something like inform the engine to navigate history items based on user interaction.

Reply:
User gesture to navigate history is what I see as "implementation details" from a dependency (GV) and will not always be true on how Gecko decides to use it.

Reply:
I think we are at an impasse. I get what you are saying, but I am also willing to accept people will have to search for those "implementation details" because there's no reasonable way to document or link to it.

Reply:
That's sort of why the suggestion is to say "informs the engine", because if we say history navigation and someone in some months from now looks, they need to go down to GV either way to see if that doc is _still_ true.

Reply:
Offline, we settled on this recommendation for now. ^